azurerm/helpers/authentication/auth_method_client_cert_test.go

```diff
@@ -0,0 +1,173 @@
+package authentication
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestServicePrincipalClientCertAuth_builder(t *testing.T) {
+	builder := Builder{
+		ClientID:           "some-client-id",
+		ClientCertPath:     "some-client-cert-path",
+		ClientCertPassword: "some-password",
+		Environment:        "some-environment",
+		SubscriptionID:     "some-subscription-id",
+		TenantID:           "some-tenant-id",
+	}
+	config := newServicePrincipalClientCertificateAuth(builder)
+	servicePrincipal := config.(servicePrincipalClientCertificateAuth)
+
+	if builder.ClientID != servicePrincipal.clientId {
+		t.Fatalf("Expected Client ID to be %q but got %q", builder.ClientID, servicePrincipal.clientId)
+	}
+
+	if builder.ClientCertPath != servicePrincipal.clientCertPath {
+		t.Fatalf("Expected Client Certificate Path to be %q but got %q", builder.ClientCertPath, servicePrincipal.clientCertPath)
+	}
+
+	if builder.ClientCertPassword != servicePrincipal.clientCertPassword {
+		t.Fatalf("Expected Client Certificate Password to be %q but got %q", builder.ClientCertPassword, servicePrincipal.clientCertPassword)
+	}
+
+	if builder.Environment != servicePrincipal.environment {
+		t.Fatalf("Expected Environment to be %q but got %q", builder.Environment, servicePrincipal.environment)
+	}
+
+	if builder.SubscriptionID != servicePrincipal.subscriptionId {
+		t.Fatalf("Expected Subscription ID to be %q but got %q", builder.SubscriptionID, servicePrincipal.subscriptionId)
+	}
+
+	if builder.TenantID != servicePrincipal.tenantId {
+		t.Fatalf("Expected Tenant ID to be %q but got %q", builder.TenantID, servicePrincipal.tenantId)
+	}
+}
+
+func TestServicePrincipalClientCertAuth_validate(t *testing.T) {
+	data := []byte("client-cert-auth")
+	filePath := "./example.pfx"
```

Comment:
Should we use `ioutil .TempFile` here to ensure no possible collisions or pollution of the project dir if something crashes?

Reply:
started out with that approach - however ultimately ended up reverting it (since the TempFile is only a prefix not a suffix) - for the reasons outlined above